packages/replay/src/util/getReplayEvent.ts

```diff
@@ -17,16 +15,20 @@ export async function getReplayEvent({
   // XXX: This event does not trigger `beforeSend` in SDK
   // @ts-ignore private api
   const preparedEvent: Event = await client._prepareEvent(event, { event_id }, scope);
-
   const session = scope && scope.getSession();
   if (session) {
     // @ts-ignore private api
     client._updateSessionFromEvent(session, preparedEvent);
   }
 
+  // extract the SDK name because `client._prepareEvent` doesn't add it to the event
+  const metadata = client.getOptions() && client.getOptions()._metadata;
+  const { name } = (metadata && metadata.sdk) || {};
+
   preparedEvent.sdk = {
```

Comment:
l: IMHO we don't need to overwrite this like that, as it will never be set. I'd just do:

```js
preparedEvent.sdk = {
  name, 
  version: 'xxx'
}
```

Or, maybe export the `getSdkMetadataForEnvelopeHeader` function or similar from core and use that somehow (but I'd say that's optional).

Reply:
If we don't use the `...preparedEvent.sdk` spread, we don't add the entire SDK metadata to the event. For instance, the integrations array won't be added. I think it's fine to omit it in the envelope header but I don't know if we should remove it from the event. 

`getSdkMetadataForEnvelopeHeader` would only make sense for constructing the envelope header. I'll take a look if this reduces bundle size

Reply:
Ok, we can use `getSdkMetadataForEnvelopeHeader`. Will do this in a follow-up PR though, as it involves moving stuff around.